Library/Homebrew/formula_installer.rb

```diff
@@ -212,6 +213,10 @@ def install
       opoo "#{formula.full_name}: #{old_flag} was deprecated; using #{new_flag} instead!"
     end
 
+    invalid_flag_names.each do |flag|
+      opoo "#{formula.full_name}: #{flag} is invalid for this formula and will be ignored!"
```

Comment:
I think ideal would be error & halt execution if no similarly named valid flag can be found. If a similarly named valid flag can be found, this should be suggested as alternative. User would then have option to choose it & continue build, or exit.